aesara/tensor/shape.py

```diff
@@ -371,22 +371,34 @@ class SpecifyShape(COp):
     def make_node(self, x, shape):
         if not isinstance(x, Variable):
             x = aet.as_tensor_variable(x)
-        shape = aet.as_tensor_variable(shape)
-        if shape.ndim > 1:
-            raise AssertionError()
-        if shape.dtype not in aesara.tensor.type.integer_dtypes:
-            raise AssertionError()
-        if isinstance(shape, TensorConstant) and shape.data.size != x.ndim:
-            raise AssertionError()
-        return Apply(self, [x, shape], [x.type()])
+        if shape == () or shape == []:
+            shape = np.array(shape, dtype=int)
+        tshape = aet.as_tensor_variable(shape)
```

Comment:
Why can't we simply use `shape = aet.as_tensor_variable(shape, ndim=1)` here?

Reply:
You're right, that's probably much better.
The option just wasn't on my radar.